src/rollup/plugins/externals-legacy.ts

```diff
@@ -139,13 +139,13 @@ export function externals(opts: NodeExternalsOptions): Plugin {
         // Absolute path, we are not sure about subpath to generate import statement
         // Guess as main subpath export
         const packageEntry = await _resolve(pkgName).catch(() => null);
-        if (packageEntry !== originalId) {
+        if (packageEntry !== id) {
```

Comment:
externals-legacy is an snapshot for old behavior let's not touch it 

```suggestion
        if (packageEntry !== originalId) {
```